server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/CrossSpaceCompactionTask.java

```diff
@@ -299,6 +299,11 @@ private boolean addReadLock(List<TsFileResource> tsFileResourceList) {
       return false;
     }
     try {
+      // Merge is not allowed when rewrite operation is running
+      if (tsFileManager.isRewriteLocked()) {
+        releaseAllLock();
+        return false;
+      }
```

Comment:
Rewrite operations may get stuck on compaction for a long time, resulting in the accumulation of unseq data in the system.